src/components/Countdown/Countdown.js

```diff
@@ -0,0 +1,167 @@
+import { h, Component } from 'preact';
+
+const COMPONENT_SIZE_TO_FONT_SIZE_RATIO = 4;
+
+class Countdown extends Component {
+    shouldComponentUpdate = () => false
+    render = () => <canvas ref={canvas => this.canvas = canvas} />
+
+    componentDidMount() {
+        this.ctx = this.canvas.getContext('2d');
+
+        // retina magic start
+        // see https://www.html5rocks.com/en/tutorials/canvas/hidpi/
+
+        const backingStoreRatio = this.ctx.webkitBackingStorePixelRatio || 1;
+        const devicePixelRatio = window.devicePixelRatio || 1;
+
+        this.ratio = devicePixelRatio / backingStoreRatio;
+        this.canvas.style.transformOrigin = '0 0';
+        this.canvas.style.transform = `scale(${1 / this.ratio}, ${1 / this.ratio})`;
+        // retina magic end
+
+        this._update(this.props);
+        this._render();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const prevProgress = this.progress;
+        this._update(nextProps);
+
+        const newProgress = this.progress;
+
+        if (nextProps.timeRemaining === nextProps.timeAmount) {
+            this._render();
+        } else {
+            this._animate({
+                from: prevProgress,
+                to: newProgress,
+                duration: 1000,
```

Comment:
`(newProgress - prevProgress) * 1000` ?

Reply:
Good point, I had this implementation initially, _but_... What if for some reason (not sure if that is possible at all) user will miss one message about clock tick (e.g. it was 40 seconds remaining, then – 38). In this case we'll have 2 secs animation which will be interrupted after next tick message. This looks not really good, I'd leave 1 sec animation for all cases

Reply:
On the other hand if we'll need to notify about timer progress more frequently than once per second we'll have troubles...

Reply:
Ideas?

Reply:
I had one, but maybe too epic for now: We update the countdown based on an interval set on the client and only use the countdown messages for correction (like atomic clock sync ⌚️ ).

Reply:
Let's assume `NORMAL_DURATION` is `1000`.
```
if (newProgress - prevProgress > NORMAL_DURATION) {
    return NORMAL_DURATION;
} else {
    return newProgress - prevProgress;
}
```

Reply:
Or.. Or... Or.....
We can store previous update time and use it as baseline for animation duration calculation. WDYT?

Reply:
The component looks over complicated already... let's keep it 1000ms for now...

Reply:
>  looks over complicated already...

That's not good 😞 I'd see what can I do in order to simplify it